hw03/hw03/hw03.Rmd

```diff
@@ -0,0 +1,109 @@
+# Homework 03
+
+You will analyze data looking at the relationship between **green reputation** and three personality traits--**compassion**, **intellectual curiosity**, and **openness to experiences**.
+The dataset includes data from **students** and **non-students**.
+
+
+```{r}
+library(tidyverse)
+library(ggplot2)
+```
+
+```{r}
+dictionary <- readr::read_csv(("green_dictionary.csv"))
+green_data <- readr::read_csv(("green_data.csv"))
+```
+
+For your assignment, do the following.
+
+  1. Inspect the item responses (e.g., with graphs or by summarizing distinct values). 
+     Is anything unusual?
+
+Get the summaries of each data file     
+```{r}
+# dictionary %>% 
+#   summarize(dictionary)
+# 
+# green_data %>% 
+# summarize(green_data)
+# fixed_green_data <- green_data %>% 
+#                 na_if("-99")
+
+
+# data_reversed <- c("green2",
+#                    "comp6",
+#                    "comp7",
+#                    "comp8",
+#                    "comp9",
+#                    "comp10",
+#                    "intel7",
+#                    "intel8",
+#                    "intel9",
+#                    "intel10",
+#                    "open7",
+#                    "open8",
+#                    "open9",
+#                    "open10")
+# 
+# recoded_green_data <- fixed_green_data %>% 
+#   mutate(across(all_of(data_reversed),
+#                 ~ recode(.x, "1"= 5,"2" = 4,"4" = 2,"5"= 1 ),
+#                 .names = "{.col}r") %>% 
+#            select(recoded_green_data))
+
+
+#Restarting everything because I couldn't figure out what was wrong.
+print(green_data)
+
+reversed <- c("green2", "comp6", "comp7", "comp8", "comp9", "comp10", 
+              "intel7", "intel8", "intel9", "intel10", "open7", "open8", 
+              "open9", "open10")
+
+green_datar <- green_data %>% 
+  mutate(across(all_of(reversed),
+                ~ recode(.x, "1"= 5,"2" = 4,"4" = 2,"5"= 1 ),
+                .names = "{.col}r"))
+
+  
+
+
+#There are -99 s in the data file that needs to be recoded
+```
+     
+  2. Compute total scores for the four scales. 
+     Recode variables as needed.
+```{r}
+green_datar %>% 
+  rowwise() %>% 
+  mutate(green_tot = sum(c_across(green1:green5), na.rm = TRUE),
+         comp_tot = sum(c_across(comp1:comp10r), na.rm = TRUE),
+         intel_tot = sum(c_across(intel1:intel10r), na.rm = TRUE),
+         open_tot = sum(c_across(open1:open10r), na.rm = TRUE))
```

Comment:
Do the recoded values replace the original values? Using a ":" to indicate a range of variables only works if the variables are consecutive. Specifically, if your recoded values are put at the end of your data set, indicating "intel1:intel10r" would take "intel1" as the first variable and "intel10r" as the last and include all variables in between.